### PR TITLE
Removed source_items_on_mixed_sources.php fixture

### PR DESCRIPTION
**Pull request by phoenix128**
### Description
Removed fixture not needed introduced by https://github.com/magento-engcom/msi/pull/1371

### Fixed Issues (if relevant)
1. magento/msi#1373: TechDebt for #1371 eliminate fixture source_items_on_mixed_sources

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/Magento/InventoryCatalog/Test/Integration/GetDefaultSourceItemBySkuTest.php [hunk 1]
```diff
@@ -39,9 +39,5 @@ public function testExecute()
             $this->getDefaultSourceItemBySku->execute('SKU-2'),
             'Unable to find default source_item for a product assigned to Default source and others'
         );
-        self::assertNull(
-            $this->getDefaultSourceItemBySku->execute('SKU-3'),
-            'Returned a wrong default source_item for a product assigned elsewhere'
-        );
```
Comment (maghamed) [inline app/code/Magento/InventoryCatalog/Test/Integration/GetDefaultSourceItemBySkuTest.php]: hey @phoenix128 , why did you remove this assertion?

Comment (phoenix128) [reply app/code/Magento/InventoryCatalog/Test/Integration/GetDefaultSourceItemBySkuTest.php]: See my new commit, I just forgot to create the new method with different fixtures
